### PR TITLE
New Resource: azurerm_bot_channel_sms

### PR DESCRIPTION
**Pull request by neil-yechenwei**
Currently, TF doesn't support Bot SMS Channel. So I submitted this PR to implement it.

Please refer this [doc](https://docs.microsoft.com/en-us/azure/bot-service/bot-service-channel-connect-twilio?view=azure-bot-service-4.0) to create SMS account for testing.

I didn't add update test case for this resource due to below issues:
1. It requires phone verification while creating the SMS account and the phone number cannot be used again while creating another SMS account. As I have only one mobile phone, so I only can create one private SMS account for testing.
2. Once the SMS account is created, this account has one month free trial and it only can bind one phone number. So I cannot create two phone numbers for testing.

--- PASS: TestAccBotChannelsRegistration/channel/smsBasic (230.10s)
--- PASS: TestAccBotChannelsRegistration/channel/smsRequiresImport (251.26s)

API Reference:
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/botservice/resource-manager/Microsoft.BotService/stable/2021-03-01/botservice.json#L1785

### CODE REVIEW

File: website/docs/r/bot_channel_sms.markdown [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+The following arguments are supported:
+
+* `resource_group_name` - (Required) The name of the resource group where the SMS Channel should be created. Changing this forces a new resource to be created.
+
+* `location` - (Required) Specifies the supported Azure location where the resource exists. Changing this forces a new resource to be created.
+
+* `bot_name` - (Required) The name of the Bot Resource this channel will be associated with. Changing this forces a new resource to be created.
+
+* `account_sid` - (Required) The account security identifier (SID) for the SMS Channel.
```
Comment (katbyte) [inline website/docs/r/bot_channel_sms.markdown:53]: lets be a bit more clear here?

```suggestion
* `sms_channel_account_security_id` - (Required) The account security identifier (SID) for the SMS Channel.
```

File: website/docs/r/bot_channel_sms.markdown [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+* `auth_token` - (Required) The authorization token for the SMS Channel.
```
Comment (katbyte) [inline website/docs/r/bot_channel_sms.markdown:55]:
```suggestion
* `sms_channel_auth_token` - (Required) The authorization token for the SMS Channel.
```

**Review by katbyte — CHANGES_REQUESTED** [review]

File: website/docs/r/bot_channel_sms.markdown (hunk 2 above)
Comment (neil-yechenwei) [reply website/docs/r/bot_channel_sms.markdown:55]: Updated

File: website/docs/r/bot_channel_sms.markdown (hunk 1 above)
Comment (neil-yechenwei) [reply website/docs/r/bot_channel_sms.markdown:53]: Updated

**Comment by neil-yechenwei** [comment]
@katbyte , thanks for your comments. I've updated code. Please have an another look. Thanks.

**Review by katbyte — APPROVED** [review]
LGTM 🌿  - just needs conflicts fixed up

**Pull request merged**